src/runtime/Instance.h

```diff
@@ -120,12 +121,20 @@ class Instance : public Object {
     const VectorWithFixedSize<Function*>& functions() { return m_functions; }
 
 private:
+    Instance(Module* module);
+    ~Instance() {}
+
+    static size_t alignedSize()
+    {
+        return (sizeof(Instance) + sizeof(void*) - 1) & ~(sizeof(void*) - 1);
+    }
+
     Module* m_module;
 
     VectorWithFixedSize<Function*> m_functions;
-    VectorWithFixedSize<Table*> m_tables;
-    VectorWithFixedSize<Memory*> m_memories;
-    VectorWithFixedSize<Global*> m_globals;
+    Memory** m_memories;
+    Table** m_tables;
+    Global** m_globals;
     VectorWithFixedSize<Tag*> m_tags;
```

Comment:
`m_functions` and `m_tags` have not beed changed.
Is this because JIT code doesn't use these data?

Reply:
Not sure yet. I can change all of them.

The basic idea is that if these things are stored in arrays, two steps are needed to get their value: `a[b[i]]`. It is generally faster and simpler if `a[i]` can be used. The interpreter could benefit from this, if the relative offset of the address would be stored in the byte code.

If you don't prefer this direction, I can drop this patch of course.

Reply:
I see. This approach seems better than before.
(Of course, we should carefully access the array to avoid out-of-range exceptions)

Are you going to update this patch more?

Reply:
I can convert m_functions and m_tags as well. I started with these three because the interpreter gets them as arguments, so I thought the rest are not used by the interpreter:
https://github.com/Samsung/walrus/blob/interp/src/interpreter/Interpreter.cpp#L217

The (fixed) size of these arrays are stored by the module, so array limit checks are possible.

Reply:
I was wrong:
https://github.com/Samsung/walrus/blob/interp/src/interpreter/Interpreter.cpp#L877

It seems the others are used as well. I will fix the code.